ee/clickhouse/queries/trends/lifecycle.py

```diff
@@ -169,3 +79,65 @@ def get_people(
         from posthog.api.person import PersonSerializer
 
         return PersonSerializer(people, many=True).data
+
+
+class LifecycleEventQuery(ClickhouseEventQuery):
+    _filter: Filter
+
+    def get_query(self):
+        date_query, date_params = self._get_date_filter()
+        self.params.update(date_params)
+
+        prop_query, prop_params = self._get_props(self._filter.properties)
+        self.params.update(prop_params)
+
+        person_query, person_params = self._get_person_query()
+        self.params.update(person_params)
+
+        groups_query, groups_params = self._get_groups_query()
+        self.params.update(groups_params)
+
+        entity_params, entity_format_params = get_entity_filtering_params(
+            self._filter.entities[0], self._team_id, table_name=self.EVENT_TABLE_ALIAS
+        )
+        self.params.update(entity_params)
+
+        return (
+            f"""
+            SELECT DISTINCT
+                person_id,
+                toDateTime(dateTrunc(%(interval)s, events.timestamp)) AS period,
+                person.created_at AS created_at
+            FROM events AS {self.EVENT_TABLE_ALIAS}
+            {self._get_distinct_id_query()}
+            {person_query}
+            {groups_query}
+            WHERE team_id = %(team_id)s
+            {entity_format_params["entity_query"]}
+            {date_query}
+            {prop_query}
+        """,
+            self.params,
+        )
+
+    @cached_property
+    def _person_query(self):
+        return ClickhousePersonQuery(self._filter, self._team_id, self._column_optimizer, extra_fields=["created_at"],)
+
+    def _get_date_filter(self):
+        _, _, date_params = parse_timestamps(filter=self._filter, team_id=self._team_id)
+        params = {**date_params, "interval": self._filter.interval}
+        # :TRICKY: We fetch all data even for the period before the graph starts up until the end of the last period
+        return (
+            f"""
+            AND timestamp >= toDateTime(dateTrunc(%(interval)s, toDateTime(%(date_from)s))) - INTERVAL 1 {self._filter.interval}
+            AND timestamp < toDateTime(dateTrunc(%(interval)s, toDateTime(%(date_to)s))) + INTERVAL 1 {self._filter.interval}
+        """,
+            params,
+        )
+
+    def _determine_should_join_distinct_ids(self):
```

Comment:
non-blocking

We specify a return type for `_determine_should_join_persons`, should we do the same for consistency here?